### PR TITLE
Reduce Ledger dependencies and move stats to FullNode

### PR DESCRIPTION
**Pull request by Geod24**
See each commit for explanation. The first two commit is not strictly related but was found while working on this, and CI isn't cheap those days.

### CODE REVIEW

**Comment by Geod24** [comment]
🟢

File: source/agora/node/Ledger.d [hunk 1]
```diff
@@ -1124,14 +1124,51 @@ public class ValidatingLedger : Ledger
...
+            data.missing_validators);
+    }
+
+    /***************************************************************************
+
+        Returns: A list of Enrollments that can be used for the next block
+
+    ***************************************************************************/
+
+    public Enrollment[] getCandidateEnrollments (in Height height) @safe
```
Comment (hewison-chris) [inline source/agora/node/Ledger.d:1055]: Yes I like the name.

File: source/agora/node/Ledger.d [hunk 2]
```diff
@@ -1145,26 +1182,27 @@ public class ValidatingLedger : Ledger
...
-                data.tx_set ~= hash;
+                result ~= hash;
 
-            if (data.tx_set.length >= max_txs)
+            if (result.length >= max_txs)
             {
-                data.tx_set.sort();
-                return;
+                result.sort();
+                return result;
```
Comment (hewison-chris) [inline source/agora/node/Ledger.d:1109]: Why not `return result.sort();` then becomes 3 lines less?

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/agora/node/Ledger.d (hunk 2 above)
Comment (Geod24) [reply source/agora/node/Ledger.d:1109]: Because the return value is a `SortedRange`.

**Review by kchulj — APPROVED** [review]
LGTM
Nice explanation.

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by Geod24** [comment]
This apparently broke the stats, PR incoming.

**Pull request merged**